### ISSUE TITLE
ACE Arsenal loses weapon attachments of weapons in backpack.

### ISSUE THREAD

**Issue by DeliciousJaffa** [acemod/ACE3#6307]
**Arma 3 Version:** `1.82.144647` stable
**CBA Version:** `3.6.1` stable
**ACE3 Version:** `3.12.2` stable

**Mods:**
```
- CBA_A3
- ace
- Other, unrelated
```

**Description:**
- Using misc items tab on backpacks in ACE Arsenal causes any stored weapons to lose their attachments. It seems ACE Arsenal is only capable of handling Magazines and Items correctly.

**Steps to reproduce:**
- Pickup a weapon into your backpack
- Enter ace arsenal
- Open backpack misc tab
- Close ACE Arsenal

**Where did the issue occur?**
- Virtual Arsenal

**Placed Modules:**
- None

**Sample loadout export**
```sqf
[
    ["arifle_MX_F","","acc_pointer_IR","optic_Aco",["30Rnd_65x39_caseless_mag",30],[],""],
    [],
    ["hgun_P07_F","","","",["16Rnd_9x21_Mag",17],[],""],
    ["U_B_CombatUniform_mcam",[["ACE_fieldDressing",2],["ACE_morphine",1],["30Rnd_65x39_caseless_mag",2,30]]],
    ["V_PlateCarrier1_rgr",[["30Rnd_65x39_caseless_mag",7,30],["16Rnd_9x21_Mag",2,17],["SmokeShell",1,1],["SmokeShellGreen",1,1],["Chemlight_green",2,1],["HandGrenade",2,1]]],
    ["B_Kitbag_mcamo",[[["arifle_MX_F","","acc_pointer_IR","optic_Aco",[],[],""],1],[["hgun_P07_F","","","",[],[],""],1]]],
    "H_HelmetB",
    "G_WirelessEarpiece_F",
    [],
    ["ItemMap","","","ItemCompass","tf_microdagr","NVGoggles"]
]
```

**Comment by dedmen** [comment]
if setUnitLoadout doesn't support it then we can't do anything about that.
BI's fault. I already have a fix for that but BI doesn't have time to copy-paste it still. Maybe after the incoming Arma hotfix.. Probably not though.

**Comment by dedmen** [comment]
Isn't arsenal using setUnitLoadout when loading loadouts? @alganthe 

What does 
>Using misc items tab on backpacks in ACE Arsenal causes any stored weapons to lose their attachments.
actually mean? clicking on the misc items tab?

**Comment by alganthe** [comment]
I know why they get stripped out of their attachments, and unfortunately I cant' fix it without introducing a bunch of other issues, there are currently no way of adding attachments to weapons in containers.

This happens when clicking the tab of a container that contains a weapon.
For example: 
- Pistol with attachments in vest
- Click vest tab, attachments are stripped.

**Comment by DeliciousJaffa** [comment]
@alganthe The way I support items in uniform/backpack containers is by editing the `getUnitLoadout` array and using `setUnitLoadout` with the transformed array.

**Comment by dedmen** [comment]
>preferably it should only happen when interacting with the stored weapon itself if setUnitLoadout isn't used

We cannot remove items from containers sadly. We have to remove all and then add them back.
And as we cannot add weapons with attachments.. There is the problem.
So as soon as you remove an item. Your weapon is goofed.

**Comment by dedmen** [comment]
Issue won't be fixed by the bot just closing it.
This is a bug, and BI's fault for not giving us a script command to do this.

**Comment by jonpas** [comment]
This would be solved by using `CBA_fnc_removeCargoX` functions. But I don't think @alganthe wants to add special handling to that.

**Comment by dedmen** [comment]
> there are currently no way of adding attachments to weapons in containers.
that is fixed now.
> This would be solved by using CBA_fnc_removeCargoX functions.
We have that now right?

**Comment by jonpas** [comment]
We had it 11 months ago as well.

**Comment by jonpas** [comment]
I believe this was fixed by #6437 and this wasn't closed, can someone confirm?
